core/src/components/cat-form-group/cat-form-group.tsx

```diff
@@ -0,0 +1,62 @@
+import { Component, Element, h, Host, Prop, Watch } from '@stencil/core';
+
+type HTMLCatFormFieldElement =
+  | HTMLCatInputElement
+  | HTMLCatTextareaElement
+  | HTMLCatSelectElement
+  | HTMLCatLabelElement;
+@Component({
+  tag: 'cat-form-group',
+  styleUrl: 'cat-form-group.scss',
+  shadow: true
+})
+export class CatFormGroup {
+  private formElements?: HTMLCatFormFieldElement[];
+
+  @Element() hostElement!: HTMLElement;
+
+  /**
+   * Whether the labels need a marker to shown if the forms fields are required or optional.<br />
+   *
+   * By default, it is set to auto, it will display the mark depending on the number of required and optional fields: <br />
+   *
+   * If there are more required, the optional will be marked.
+   * If there are less required, it will mark the required.
+   */
+  @Prop() requiredMarker: 'none' | 'required' | 'optional' | 'auto' = 'auto';
+
+  @Watch('requiredMarker')
+  onRequiredMarker(newRequiredMarker: 'none' | 'required' | 'optional' | 'auto') {
+    let updateMarker;
+    if (this.formElements) {
+      updateMarker = newRequiredMarker === 'auto' ? this.calculate(this.formElements) : newRequiredMarker;
+      for (const element of this.formElements) {
+        if (!element.requiredMarker) {
+          element.requiredMarker = updateMarker;
+        }
+      }
+    }
+  }
+
+  componentDidLoad(): void {
+    this.formElements = Array.from(
```

Comment:
for now, this is ok, but I guess we need to optimize this at some point because often some form elements are dynamically added using ngIf. Then this breaks